### PR TITLE
Clean up and fix minor bugs in Grunt code

### PR DESCRIPTION
**Pull request by brianhelba**
* Upgrade grunt-shell and cleanup its usage for building docs
* Reorganize the Grunt Fontello building, ensuring correct task ordering
* Roll the "init" Grunt task into the "default" Grunt task
* Rename the "setup" task to "local-config", and clean it up
* Upgrade grunt-contrib-uglify

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -98,33 +95,19 @@ module.exports = function (grunt) {
...
      *     'uglify:a': {
      *       'dependencies': ['copy:a']
      *     }
      *   }
      * }
      *
-     * The init task will run `copy:a` followed by `uglify:a`.
+     * The 'default' task will run `copy:a` followed by `uglify:a`.
      */
-    grunt.registerTask('init', sortTasks(grunt.config.get('init')));
```
Comment (jbeezley) [inline Gruntfile.js]: I don't know of any examples of this, but it's possible there might be plugins that rely on the init task.  What do you think about replacing this task with a deprecation warning?  I think you could also merge the `init` object with the `default` object before running the next line.

Comment (brianhelba) [inline Gruntfile.js]: Sounds reasonable, will do.

Comment (brianhelba) [inline Gruntfile.js]: @jbeezley Done.

File: grunt_tasks/localConfig.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+module.exports = function (grunt) {
+    const confDir = `${grunt.config.get('girderDir')}/conf`;
+    const localConfigPath = `${confDir}/girder.local.cfg`;
+
+    grunt.config.merge({
+        copy: {
+            'local-config': {
+                src: `${confDir}/girder.dist.cfg`,
+                dest: localConfigPath
+            }
```
Comment (jbeezley) [inline grunt_tasks/localConfig.js]: I don't know about doing this as part of the client build.  It is really a server issue.  I think a more appropriate place to create this file automatically would be as part of the python install.

Comment (zachmullen) [inline grunt_tasks/localConfig.js]: Agreed, this should not be part of the web client build.

Comment (brianhelba) [inline grunt_tasks/localConfig.js]: Yeah, I was really surprised to see it here too, and agree that it should live in the Python code.

However, this PR is just a refactor / bugfix of some of existing functionality, so moving the functionality is a bit out of scope. I've created an issue here: https://github.com/girder/girder/issues/2263 , which may be good for @sutartmelson or @Pierre-Assemat to handle.

Comment (jbeezley) [inline grunt_tasks/localConfig.js]: That works for me.

**Pull request merged**